conda_forge_tick/update_upstream_versions.py

```diff
@@ -170,34 +173,80 @@ def get_latest_version(
     return version_data
 
 
-def get_latest_version_containerized(name, attrs, sources):
-    if sources != all_version_sources():
-        warnings.warn(
-            "The sources argument is ignored when running in a container. "
-            "All available sources will be used.",
-        )
+def get_latest_version_containerized(
+    name: str,
+    attrs: Mapping[str, Any],
+    sources: Iterable[AbstractSource],
+) -> Dict[str, Union[Literal[False], str]]:
+    """
+    Given a package, return the new version information to be written into the cf-graph.
 
+    **This function runs the version parsing in a container.**
+
+    Parameters
+    ----------
+    name : str
+        The name of the feedstock.
+    attrs : Mapping[str, Any]
+        The node attributes of the feedstock.
+    sources : Iterable[AbstractSource]
+        The version sources to use.
+
+    Returns
+    -------
+    version_data : dict
+        The new version information.
+    """
     if "feedstock_name" not in attrs:
         attrs["feedstock_name"] = name
 
-    cmd = [
-        *get_default_container_run_args(),
-        "-t",
-        get_default_container_name(),
-        "python",
-        "/opt/autotick-bot/docker/run_bot_task.py",
-        "update-version",
+    task = "get-latest-version"
+    args = [
         "--existing-feedstock-node-attrs",
         dumps(attrs.data) if isinstance(attrs, LazyJson) else dumps(attrs),
+        "--sources",
+        ",".join([source.name for source in sources]),
     ]
 
-    res = subprocess.run(
-        cmd,
-        capture_output=True,
-    )
-    if res.returncode != 0:
-        raise RuntimeError(f"Error running containerized version update: {res.stderr}")
-    return json.loads(res.stdout.decode("utf-8"))
+    return run_container_task(task, args)
+
+
+def get_latest_version(
+    name: str,
+    attrs: Mapping[str, Any],
+    sources: Iterable[AbstractSource],
+    use_container: bool = False,
+) -> Dict[str, Union[Literal[False], str]]:
+    """
+    Given a package, return the new version information to be written into the cf-graph.
+
+    Parameters
+    ----------
+    name : str
+        The name of the feedstock.
+    attrs : Mapping[str, Any]
+        The node attributes of the feedstock.
+    sources : Iterable[AbstractSource]
+        The version sources to use.
+    use_container : bool, optional
+        Whether to use a container to run the version parsing.
+        If None, the function will use a container if the environment
+        variable `CF_TICK_IN_CONTAINER` is 'false'. This feature can be
```

Comment:
Is this accurate? My first impression that use_container would be True if CF_TICK_IN_CONTAINER is truthy. But I guess this env var means that "we are already in a container, do not start another one", right?

Reply:
Yes. This is to prevent docker in docker.